cmd/helm/push_test.go

```diff
@@ -17,11 +17,138 @@ limitations under the License.
 package main
 
 import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
 	"testing"
+	"time"
+
+	"helm.sh/helm/v3/pkg/action"
+	helmRegistry "helm.sh/helm/v3/pkg/registry"
+
+	"github.com/distribution/distribution/v3/configuration"
+	"github.com/distribution/distribution/v3/registry"
+	"github.com/phayes/freeport"
+	"github.com/stretchr/testify/suite"
 )
 
 func TestPushFileCompletion(t *testing.T) {
 	checkFileCompletion(t, "push", true)
 	checkFileCompletion(t, "push package.tgz", false)
 	checkFileCompletion(t, "push package.tgz oci://localhost:5000", false)
 }
+
+var (
+	testWorkspaceDir = "helm-registry-test"
+)
+
+type HelmPushTestSuite struct {
+	suite.Suite
+	Out                     io.Writer
+	DockerRegistryHost      string
+	CompromisedRegistryHost string
+	WorkspaceDir            string
+	RegistryClient          *helmRegistry.Client
+	PushClient              *action.Push
+}
+
+func getClientIP() (string, error) {
+	addrs, err := net.InterfaceAddrs()
+
+	if err != nil {
+		return "", err
+	}
+	for _, address := range addrs {
+		// filter loopback ip
+		if ipNet, ok := address.(*net.IPNet); ok && !ipNet.IP.IsLoopback() {
+			if ipNet.IP.To4() != nil {
+				return ipNet.IP.String(), nil
+			}
+		}
+	}
+
+	return "", errors.New("can not find the client ip address")
+
+}
+
+func (suite *HelmPushTestSuite) SetupSuite() {
+	suite.WorkspaceDir = testWorkspaceDir
+	os.RemoveAll(suite.WorkspaceDir)
+	os.Mkdir(suite.WorkspaceDir, 0700)
+
+	var out bytes.Buffer
+	suite.Out = &out
+	//credentialsFile := filepath.Join(suite.WorkspaceDir, registry.CredentialsFileBasename)
```

Comment:
if the code is useless. please remove it.

Reply:
This variable is initialized in `SetupSuite` func and used in `Test_1_Push` and `Test_2_Push` test func

Reply:
//credentialsFile := filepath.Join(suite.WorkspaceDir, registry.CredentialsFileBasename)

Reply:
yes，You are right